yajsapi/agreement/agreement.ts

```diff
@@ -0,0 +1,80 @@
+import { Logger } from "../utils";
+import { RequestorApi } from "ya-ts-client/dist/ya-market/api";
+import { Agreement as AgreementModel, AgreementStateEnum } from "ya-ts-client/dist/ya-market/src/models";
+import { AgreementConfigContainer } from "./agreement_config_container";
+
+export { AgreementStateEnum };
+
+export interface AgreementOptions {
+  credentials?: { apiKey?: string; basePath?: string };
+  requestTimeout?: number;
+  executeTimeout?: number;
+  eventPoolingInterval?: number;
+  eventPoolingMaxEventsPerRequest?: number;
+  logger?: Logger;
+}
+
+export interface ProviderInfo {
+  providerName: string;
+  providerId: string | null;
+}
+
+export class Agreement {
+  private readonly api: RequestorApi;
+  private readonly logger?: Logger;
+  private readonly requestTimeout: number;
+
+  private agreementData?: AgreementModel;
+
+  constructor(public readonly id, private readonly configContainer: AgreementConfigContainer) {
+    this.logger = configContainer.logger;
+    this.api = configContainer.api;
+    this.requestTimeout = configContainer.options?.requestTimeout || 10000;
+  }
+
+  async refreshDetails() {
+    const { data } = await this.api.getAgreement(this.id, { timeout: this.requestTimeout });
+    this.agreementData = data;
+  }
+
+  getProviderInfo(): ProviderInfo {
+    return {
+      providerName: this.agreementData?.offer?.properties["golem.node.id.name"] || null,
+      providerId: this.agreementData?.offer?.providerId || null,
+    };
+  }
+
+  async getState(): Promise<AgreementStateEnum> {
+    await this.refreshDetails();
+    return this.agreementData!.state;
+  }
+
+  async isFinalState(): Promise<boolean> {
```

Comment:
I wonder if it should not be on the service? (In the context of using mid-level standalone) Only service knows what finalState is... ?